eventheader/src/native.rs

```diff
@@ -89,118 +96,125 @@ impl UserEventsDataFile {
         {
             // Need to find the ".../tracing/user_events_data" file in tracefs or debugfs.
 
-            // Determine tracefs/debugfs mount point by parsing "/proc/mounts".
-            // Prefer "tracefs" over "debugfs": if we find a debugfs, save the path but
-            // keep looking in case we find a tracefs later.
-            clear_errno();
-            let mounts_file = unsafe {
-                linux::fopen(
-                    "/proc/mounts\0".as_ptr().cast::<ffi::c_char>(),
-                    "r\0".as_ptr().cast::<ffi::c_char>(),
-                )
-            };
-            if mounts_file.is_null() {
-                new_file_or_error = -get_failure_errno();
+            // First, try the normal tracefs/debugfs mount points.
+            if let new_file @ 0.. = open_rdwr(b"/sys/kernel/tracing/user_events_data\0") {
```

Comment:
What's this "@ 0.." mean?

Reply:
The `@` means "where it matches". The result is similar to C++ `if (auto new_file = open_rdwr(); new_file >= 0)`.